hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java

```diff
@@ -132,9 +132,10 @@ protected HazelcastInstanceImpl(String name, Config config, NodeContext nodeCont
 
         try {
             logger = node.getLogger(getClass().getName());
-            lifecycleService.fireLifecycleEvent(STARTING);
```

Comment:
yes, that would be ideal. at least this is [documented](http://docs.hazelcast.org/docs/3.7/manual/html-single/index.html#implementing-hazelcastinstanceaware): 
`[...] The reason is that when HazelcastInstance is injected for a HazelcastInstanceAware implementation, it may not be up and running at the injection time.`